Automattic/sensei
# Package exported CSVs into a ZIP file when possible

Fixes #3512

### Changes proposed in this Pull Request

* This currently just handles the actual packaging. Once it has packaged the CSVs and checks that it is successful, it will delete the old files and return the ZIP to the client.
* We should probably rephrase the wording in the final step before we list the files. It will usually be a ZIP file containing the CSVs, but if for some reason `ZipArchive` isn't included in their PHP build (rare; so much so that WP core uses it for privacy functionality, but it does check it exists first).
* I'm not doing any logging because it didn't seem to be used in the rest of the export job. For now, I'm having it fail gracefully to just show the CSVs if the ZIP file couldn't be generated.
* Slight difference from the issue: I'm appending the date onto the ZIP filename. This is still inconsistent with the CSV filenames (based on site name), but thought I'd stick close to issue for now.

### Testing instructions

* Run the exporter. Be presented with a ZIP file.
* ZIP file should contain the CSVs as presented previously (same filenames).

<img width="716" alt="Screen Shot 2020-08-12 at 6 26 46 pm" src="https://user-images.githubusercontent.com/68693/90047171-62a8fa80-dcc9-11ea-98ea-6d79828dac16.png">
(Probably should update the wording)
